### DISCUSSION TITLE
How to define a slash command that I can use in the chat?

### DISCUSSION THREAD

**Discussion by dlants** [olimorris/codecompanion.nvim#300] [Q&A] [answered]
I want to define some commands, like "/libcontext" that I can use to dump some stuff into the context for the LLM. I think this should be possible (the readme advertises "custom slash commands") but I can't seem to find any docs or recipes for how to do it, or any dotfiles online that show an example of this.

I found the  [slash_commands](https://github.com/olimorris/codecompanion.nvim/blob/main/lua/codecompanion/config.lua#L53) option but that requires a callback that seems to always be a string. I also used the `prompt_library` option, like so:

```
prompt_library = {
            ["libcontext"] = {
              strategy = "chat",
              description = "test",
              opts = {
                slash_cmd = "libcontext"
              },
              prompts = {
                {
                  role = "system",
                  content = function()
                    return "some context"
                  end
                }
              }
            }
```

But that prompt doesn't seem to show up when I type `/` into the chat.
```
## Me ─

What does this say? :
/libcontext



## CodeCompanion ─

This appears to be a Unix-style directory or file path. It likely refers to a directory named "libcontext" at the root level of a file system. However, without more context, it's hard to be certain about its exact meaning or purpose. It could potentially be:
```

I am able to use the prompt via exec mode: `:CodeCompanion /libcontext` correctly opens the chat and populates the text.

Thanks!

**Comment by olimorris** [comment, marked as answer]
I'm actually working on a feature that will enable you to have a prompt library appear as a slash command. Expecting it to be finished in the next couple of days.

Right now you could define your own slash command like we have in the helpers folder.

**Reply by dlants** [reply]
cool, thanks for the prompt response. Since you're working on it (I'm assuming it's the `/prompt` command), it would be really nice if the prompt is folded when it's inserted into the chat so it doesn't take up the entire window!

**Reply by dlants** [reply]
For defining your own slash commands, I see that the `callback` in the example config is always a string like `helpers.slash_commands.*`. Can that param be a function?

**Reply by olimorris** [reply]
> For defining your own slash commands, I see that the `callback` in the example config is always a string like `helpers.slash_commands.*`. Can that param be a function?

Yep great shout. You can currently point to a file in your config which returns a function.

**Reply by olimorris** [reply]
> cool, thanks for the prompt response. Since you're working on it (I'm assuming it's the `/prompt` command), it would be really nice if the prompt is folded when it's inserted into the chat so it doesn't take up the entire window!

Another feature I'm mulling over is the hiding of any data in the chat buffer which the user hasn't typed. It seems messy to be to have slash command and prompt library output in the buffer. So may do away with the folding completely.

**Comment by olimorris** [comment]
This should now be available as part of the [v8.0.0](https://github.com/olimorris/codecompanion.nvim/discussions/304) release.
